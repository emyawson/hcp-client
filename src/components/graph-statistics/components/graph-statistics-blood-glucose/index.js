export * from './graph-statistics-blood-glucose.component';
