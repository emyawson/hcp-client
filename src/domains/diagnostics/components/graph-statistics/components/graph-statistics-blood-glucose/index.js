export {
  GraphDetailBloodGlucose,
} from './graph-statistics-blood-glucose.component';
