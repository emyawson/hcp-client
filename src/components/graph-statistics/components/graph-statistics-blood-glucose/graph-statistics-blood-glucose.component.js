import React from 'react';

import { CrossMarkIcon } from 'src/assets/icons';
import { LocalizedText } from 'src/components';

import {
  IconContainer,
  ListSection,
  ListsContainer,
  List,
  ListElement,
  ListElementSectionHeader,
} from '../../graph-statistics.style';

type GraphDetailBloodGlucoseProps = {
  bloodGlucoseMean: number,
  bloodGlucoseStandardDeviation: number,
  testsPerDay: number,
};

export const GraphDetailBloodGlucose = ({
  bloodGlucoseMean,
  bloodGlucoseStandardDeviation,
  testsPerDay,
}: GraphDetailBloodGlucoseProps) => (
  <ListSection borderRight>
    <ListElementSectionHeader>
      <IconContainer>
        <CrossMarkIcon />{' '}
      </IconContainer>
      <LocalizedText textKey="graphDetails.statistics.bloodGlucose.meanBloodGlucoseTitle" />
    </ListElementSectionHeader>
    <ListsContainer>
      <List flex={'none'}>
        <ListElement bold>
          <LocalizedText textKey="graphDetails.statistics.bloodGlucose.glucoseLevel" />
        </ListElement>
        <ListElement bold>
          <LocalizedText textKey="graphDetails.statistics.bloodGlucose.standardDeviation" />
        </ListElement>
        <ListElement bold>
          <LocalizedText textKey="graphDetails.statistics.bloodGlucose.testsPerDay" />
        </ListElement>
      </List>
      <List>
        <ListElement>
          {`${bloodGlucoseMean || '-'} `}
          <LocalizedText textKey="graphDetails.mgPerDL" />
        </ListElement>
        <ListElement>
          {`${bloodGlucoseStandardDeviation} `}
          <LocalizedText textKey="graphDetails.mgPerDL" />
        </ListElement>
        <ListElement>{testsPerDay}</ListElement>
      </List>
      <List flex={8} />
    </ListsContainer>
  </ListSection>
);
