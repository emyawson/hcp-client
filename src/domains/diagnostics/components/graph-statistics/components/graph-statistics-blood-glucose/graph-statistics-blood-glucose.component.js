import React from 'react';

import { CrossMarkIcon } from 'src/domains/diagnostics/assets/icons';
import { LocalizedText } from 'src/domains/diagnostics/components';

import {
  IconContainer,
  ListSection,
  ListsContainer,
  List,
  ListElement,
  ListElementSectionHeader,
} from '../../graph-statistics.style';

export const GraphDetailBloodGlucose = ({
  bloodGlucoseMean,
  bloodGlucoseStandardDeviation,
  testsPerDay,
}) => (
  <ListSection borderRight>
    <ListElementSectionHeader>
      <IconContainer>
        <CrossMarkIcon />{' '}
      </IconContainer>
      <LocalizedText textKey="graphDetails.statistics.bloodGlucose.meanBloodGlucoseTitle" />
    </ListElementSectionHeader>
    <ListsContainer>
      <List flex={'none'}>
        <ListElement bold>
          <LocalizedText textKey="graphDetails.statistics.bloodGlucose.glucoseLevel" />
        </ListElement>
        <ListElement bold>
          <LocalizedText textKey="graphDetails.statistics.bloodGlucose.standardDeviation" />
        </ListElement>
        <ListElement bold>
          <LocalizedText textKey="graphDetails.statistics.bloodGlucose.testsPerDay" />
        </ListElement>
      </List>
      <List>
        <ListElement>
          {`${bloodGlucoseMean || '-'} `}
          <LocalizedText textKey="graphDetails.mgPerDL" />
        </ListElement>
        <ListElement>
          {`${bloodGlucoseStandardDeviation} `}
          <LocalizedText textKey="graphDetails.mgPerDL" />
        </ListElement>
        <ListElement>{testsPerDay}</ListElement>
      </List>
      <List flex={8} />
    </ListsContainer>
  </ListSection>
);
