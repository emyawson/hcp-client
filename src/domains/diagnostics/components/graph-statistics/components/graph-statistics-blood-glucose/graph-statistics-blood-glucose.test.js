import React from 'react';
import { shallow } from 'enzyme';

import { GraphDetailBloodGlucose } from './graph-statistics-blood-glucose.component';

describe('graph-statistics-blood-glucose test suite', () => {
  test('it renders correctly', () => {
    const wrapper = shallow(
      <GraphDetailBloodGlucose
        bloodGlucoseMean={100}
        bloodGlucoseStandardDeviation={85}
        testsPerDay={12}
      />,
    );
    expect(wrapper).toMatchSnapshot();
  });
});
