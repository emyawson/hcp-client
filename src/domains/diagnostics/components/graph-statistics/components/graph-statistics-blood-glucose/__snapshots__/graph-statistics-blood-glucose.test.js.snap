// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`graph-statistics-blood-glucose test suite it renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    bloodGlucoseMean={100}
    bloodGlucoseStandardDeviation={85}
    testsPerDay={12}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "class",
    "props": Object {
      "borderRight": true,
      "children": Array [
        <styled.div>
          <styled.div>
            <Unknown />
             
          </styled.div>
          <LocalizedText
            textKey="graphDetails.statistics.bloodGlucose.meanBloodGlucoseTitle"
          />
        </styled.div>,
        <styled.div>
          <styled.ul
            flex="none"
          >
            <styled.li
              bold={true}
            >
              <LocalizedText
                textKey="graphDetails.statistics.bloodGlucose.glucoseLevel"
              />
            </styled.li>
            <styled.li
              bold={true}
            >
              <LocalizedText
                textKey="graphDetails.statistics.bloodGlucose.standardDeviation"
              />
            </styled.li>
            <styled.li
              bold={true}
            >
              <LocalizedText
                textKey="graphDetails.statistics.bloodGlucose.testsPerDay"
              />
            </styled.li>
          </styled.ul>
          <styled.ul>
            <styled.li>
              100 
              <LocalizedText
                textKey="graphDetails.mgPerDL"
              />
            </styled.li>
            <styled.li>
              85 
              <LocalizedText
                textKey="graphDetails.mgPerDL"
              />
            </styled.li>
            <styled.li>
              12
            </styled.li>
          </styled.ul>
          <styled.ul
            flex={8}
          />
        </styled.div>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": null,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <styled.div>
              <Unknown />
               
            </styled.div>,
            <LocalizedText
              textKey="graphDetails.statistics.bloodGlucose.meanBloodGlucoseTitle"
            />,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": null,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                <Unknown />,
                " ",
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": null,
                "nodeType": "function",
                "props": Object {},
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              " ",
            ],
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": null,
            "nodeType": "function",
            "props": Object {
              "textKey": "graphDetails.statistics.bloodGlucose.meanBloodGlucoseTitle",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": null,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <styled.ul
              flex="none"
            >
              <styled.li
                bold={true}
              >
                <LocalizedText
                  textKey="graphDetails.statistics.bloodGlucose.glucoseLevel"
                />
              </styled.li>
              <styled.li
                bold={true}
              >
                <LocalizedText
                  textKey="graphDetails.statistics.bloodGlucose.standardDeviation"
                />
              </styled.li>
              <styled.li
                bold={true}
              >
                <LocalizedText
                  textKey="graphDetails.statistics.bloodGlucose.testsPerDay"
                />
              </styled.li>
            </styled.ul>,
            <styled.ul>
              <styled.li>
                100 
                <LocalizedText
                  textKey="graphDetails.mgPerDL"
                />
              </styled.li>
              <styled.li>
                85 
                <LocalizedText
                  textKey="graphDetails.mgPerDL"
                />
              </styled.li>
              <styled.li>
                12
              </styled.li>
            </styled.ul>,
            <styled.ul
              flex={8}
            />,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": null,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                <styled.li
                  bold={true}
                >
                  <LocalizedText
                    textKey="graphDetails.statistics.bloodGlucose.glucoseLevel"
                  />
                </styled.li>,
                <styled.li
                  bold={true}
                >
                  <LocalizedText
                    textKey="graphDetails.statistics.bloodGlucose.standardDeviation"
                  />
                </styled.li>,
                <styled.li
                  bold={true}
                >
                  <LocalizedText
                    textKey="graphDetails.statistics.bloodGlucose.testsPerDay"
                  />
                </styled.li>,
              ],
              "flex": "none",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": null,
                "nodeType": "class",
                "props": Object {
                  "bold": true,
                  "children": <LocalizedText
                    textKey="graphDetails.statistics.bloodGlucose.glucoseLevel"
                  />,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": null,
                  "nodeType": "function",
                  "props": Object {
                    "textKey": "graphDetails.statistics.bloodGlucose.glucoseLevel",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": null,
                "nodeType": "class",
                "props": Object {
                  "bold": true,
                  "children": <LocalizedText
                    textKey="graphDetails.statistics.bloodGlucose.standardDeviation"
                  />,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": null,
                  "nodeType": "function",
                  "props": Object {
                    "textKey": "graphDetails.statistics.bloodGlucose.standardDeviation",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": null,
                "nodeType": "class",
                "props": Object {
                  "bold": true,
                  "children": <LocalizedText
                    textKey="graphDetails.statistics.bloodGlucose.testsPerDay"
                  />,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": null,
                  "nodeType": "function",
                  "props": Object {
                    "textKey": "graphDetails.statistics.bloodGlucose.testsPerDay",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                "type": [Function],
              },
            ],
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": null,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                <styled.li>
                  100 
                  <LocalizedText
                    textKey="graphDetails.mgPerDL"
                  />
                </styled.li>,
                <styled.li>
                  85 
                  <LocalizedText
                    textKey="graphDetails.mgPerDL"
                  />
                </styled.li>,
                <styled.li>
                  12
                </styled.li>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": null,
                "nodeType": "class",
                "props": Object {
                  "children": Array [
                    "100 ",
                    <LocalizedText
                      textKey="graphDetails.mgPerDL"
                    />,
                  ],
                },
                "ref": null,
                "rendered": Array [
                  "100 ",
                  Object {
                    "instance": null,
                    "key": null,
                    "nodeType": "function",
                    "props": Object {
                      "textKey": "graphDetails.mgPerDL",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                ],
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": null,
                "nodeType": "class",
                "props": Object {
                  "children": Array [
                    "85 ",
                    <LocalizedText
                      textKey="graphDetails.mgPerDL"
                    />,
                  ],
                },
                "ref": null,
                "rendered": Array [
                  "85 ",
                  Object {
                    "instance": null,
                    "key": null,
                    "nodeType": "function",
                    "props": Object {
                      "textKey": "graphDetails.mgPerDL",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                ],
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": null,
                "nodeType": "class",
                "props": Object {
                  "children": 12,
                },
                "ref": null,
                "rendered": 12,
                "type": [Function],
              },
            ],
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": null,
            "nodeType": "class",
            "props": Object {
              "flex": 8,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "class",
      "props": Object {
        "borderRight": true,
        "children": Array [
          <styled.div>
            <styled.div>
              <Unknown />
               
            </styled.div>
            <LocalizedText
              textKey="graphDetails.statistics.bloodGlucose.meanBloodGlucoseTitle"
            />
          </styled.div>,
          <styled.div>
            <styled.ul
              flex="none"
            >
              <styled.li
                bold={true}
              >
                <LocalizedText
                  textKey="graphDetails.statistics.bloodGlucose.glucoseLevel"
                />
              </styled.li>
              <styled.li
                bold={true}
              >
                <LocalizedText
                  textKey="graphDetails.statistics.bloodGlucose.standardDeviation"
                />
              </styled.li>
              <styled.li
                bold={true}
              >
                <LocalizedText
                  textKey="graphDetails.statistics.bloodGlucose.testsPerDay"
                />
              </styled.li>
            </styled.ul>
            <styled.ul>
              <styled.li>
                100 
                <LocalizedText
                  textKey="graphDetails.mgPerDL"
                />
              </styled.li>
              <styled.li>
                85 
                <LocalizedText
                  textKey="graphDetails.mgPerDL"
                />
              </styled.li>
              <styled.li>
                12
              </styled.li>
            </styled.ul>
            <styled.ul
              flex={8}
            />
          </styled.div>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": null,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <styled.div>
                <Unknown />
                 
              </styled.div>,
              <LocalizedText
                textKey="graphDetails.statistics.bloodGlucose.meanBloodGlucoseTitle"
              />,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": null,
              "nodeType": "class",
              "props": Object {
                "children": Array [
                  <Unknown />,
                  " ",
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": null,
                  "nodeType": "function",
                  "props": Object {},
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                " ",
              ],
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": null,
              "nodeType": "function",
              "props": Object {
                "textKey": "graphDetails.statistics.bloodGlucose.meanBloodGlucoseTitle",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <styled.ul
                flex="none"
              >
                <styled.li
                  bold={true}
                >
                  <LocalizedText
                    textKey="graphDetails.statistics.bloodGlucose.glucoseLevel"
                  />
                </styled.li>
                <styled.li
                  bold={true}
                >
                  <LocalizedText
                    textKey="graphDetails.statistics.bloodGlucose.standardDeviation"
                  />
                </styled.li>
                <styled.li
                  bold={true}
                >
                  <LocalizedText
                    textKey="graphDetails.statistics.bloodGlucose.testsPerDay"
                  />
                </styled.li>
              </styled.ul>,
              <styled.ul>
                <styled.li>
                  100 
                  <LocalizedText
                    textKey="graphDetails.mgPerDL"
                  />
                </styled.li>
                <styled.li>
                  85 
                  <LocalizedText
                    textKey="graphDetails.mgPerDL"
                  />
                </styled.li>
                <styled.li>
                  12
                </styled.li>
              </styled.ul>,
              <styled.ul
                flex={8}
              />,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": null,
              "nodeType": "class",
              "props": Object {
                "children": Array [
                  <styled.li
                    bold={true}
                  >
                    <LocalizedText
                      textKey="graphDetails.statistics.bloodGlucose.glucoseLevel"
                    />
                  </styled.li>,
                  <styled.li
                    bold={true}
                  >
                    <LocalizedText
                      textKey="graphDetails.statistics.bloodGlucose.standardDeviation"
                    />
                  </styled.li>,
                  <styled.li
                    bold={true}
                  >
                    <LocalizedText
                      textKey="graphDetails.statistics.bloodGlucose.testsPerDay"
                    />
                  </styled.li>,
                ],
                "flex": "none",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": null,
                  "nodeType": "class",
                  "props": Object {
                    "bold": true,
                    "children": <LocalizedText
                      textKey="graphDetails.statistics.bloodGlucose.glucoseLevel"
                    />,
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": null,
                    "nodeType": "function",
                    "props": Object {
                      "textKey": "graphDetails.statistics.bloodGlucose.glucoseLevel",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": null,
                  "nodeType": "class",
                  "props": Object {
                    "bold": true,
                    "children": <LocalizedText
                      textKey="graphDetails.statistics.bloodGlucose.standardDeviation"
                    />,
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": null,
                    "nodeType": "function",
                    "props": Object {
                      "textKey": "graphDetails.statistics.bloodGlucose.standardDeviation",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": null,
                  "nodeType": "class",
                  "props": Object {
                    "bold": true,
                    "children": <LocalizedText
                      textKey="graphDetails.statistics.bloodGlucose.testsPerDay"
                    />,
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": null,
                    "nodeType": "function",
                    "props": Object {
                      "textKey": "graphDetails.statistics.bloodGlucose.testsPerDay",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                  "type": [Function],
                },
              ],
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": null,
              "nodeType": "class",
              "props": Object {
                "children": Array [
                  <styled.li>
                    100 
                    <LocalizedText
                      textKey="graphDetails.mgPerDL"
                    />
                  </styled.li>,
                  <styled.li>
                    85 
                    <LocalizedText
                      textKey="graphDetails.mgPerDL"
                    />
                  </styled.li>,
                  <styled.li>
                    12
                  </styled.li>,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": null,
                  "nodeType": "class",
                  "props": Object {
                    "children": Array [
                      "100 ",
                      <LocalizedText
                        textKey="graphDetails.mgPerDL"
                      />,
                    ],
                  },
                  "ref": null,
                  "rendered": Array [
                    "100 ",
                    Object {
                      "instance": null,
                      "key": null,
                      "nodeType": "function",
                      "props": Object {
                        "textKey": "graphDetails.mgPerDL",
                      },
                      "ref": null,
                      "rendered": null,
                      "type": [Function],
                    },
                  ],
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": null,
                  "nodeType": "class",
                  "props": Object {
                    "children": Array [
                      "85 ",
                      <LocalizedText
                        textKey="graphDetails.mgPerDL"
                      />,
                    ],
                  },
                  "ref": null,
                  "rendered": Array [
                    "85 ",
                    Object {
                      "instance": null,
                      "key": null,
                      "nodeType": "function",
                      "props": Object {
                        "textKey": "graphDetails.mgPerDL",
                      },
                      "ref": null,
                      "rendered": null,
                      "type": [Function],
                    },
                  ],
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": null,
                  "nodeType": "class",
                  "props": Object {
                    "children": 12,
                  },
                  "ref": null,
                  "rendered": 12,
                  "type": [Function],
                },
              ],
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": null,
              "nodeType": "class",
              "props": Object {
                "flex": 8,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
